course/submission.go

```diff
@@ -0,0 +1,404 @@
+	).All(context.Background(), db)
+	if err != nil {
+		return nil, err
+	}
+
+	return submissions, nil
+}
+
+func GetUserSubmission(db *sql.DB, user_submission_id int) (*models.UserSubmission, error) {
+	user_submission, err := models.FindUserSubmission(context.Background(), db, user_submission_id)
+	if err != nil {
+		return nil, err
+	}
+	return user_submission, nil
+}
+
+func CreateUserSubmission(db *sql.DB, name string, submitter_id int, submission_id int, ignores_submission_deadline int8) (int, error) {
+	var nullname null.String
+	curtime := time.Now()
+	if name == "" {
+		nullname = null.NewString(name, false)
+	} else {
+		nullname = null.NewString(name, true)
+	}
+	if ignores_submission_deadline == 0 {
+		subm, err := models.FindSubmission(context.Background(), db, submission_id)
+		if err != nil {
+			return 0, err
+		}
+
+		if subm.Deadline.Time.Sub(curtime) < 0 {
+			return 0, errors.New("past Deadline time")
+		}
+	}
+	tx, err := db.BeginTx(context.Background(), nil)
+	if err != nil {
+		return 0, err
+	}
+
+	uhassubmission := models.UserSubmission{Name: nullname, SubmitterID: submitter_id, SubmissionID: submission_id, IgnoresSubmissionDeadline: ignores_submission_deadline, SubmissionTime: null.NewTime(curtime, true)}
+
+	err = uhassubmission.Insert(context.Background(), tx, boil.Infer())
+	if err != nil {
+		if e := tx.Rollback(); e != nil {
+			return 0, fmt.Errorf("fatal: unable to rollback transaction on error: %s; %s", err.Error(), e.Error())
+		}
+		return 0, err
+	}
+
+	if e := tx.Commit(); e != nil {
+		return 0, fmt.Errorf("fatal: unable to commit transaction on error: %s; %s", err, e)
+	}
+
+	return uhassubmission.ID, nil
+}
+
+/*
+func EditUserSubmission(db *sql.DB, user_submission_id int, file_id int, name string) (int, error) {
+	tx, err := db.BeginTx(context.Background(), nil)
+	if err != nil {
+		return 0, err
+	}
+	uhassubmission, err := models.FindUserSubmission(context.Background(), db, user_submission_id)
+	if err != nil {
+		return 0, err
+	}
+	if name == "" {
+		uhassubmission.Name = null.NewString(name, false)
+	} else {
+		uhassubmission.Name = null.NewString(name, true)
+	}
+
+	_, err = uhassubmission.Update(context.Background(), tx, boil.Infer())
+	if e := tx.Commit(); e != nil {
+		return 0, fmt.Errorf("fatal: unable to commit transaction on error: %s; %s", err, e)
+	}
+	file, err := models.FindFile(context.Background(), tx, file_id)
+	if err != nil {
+		if e := tx.Rollback(); e != nil {
+			return 0, fmt.Errorf("fatal: unable to rollback transaction on error: %s; %s", err.Error(), e.Error())
+		}
+	}
+	file.Name = name
+	_, err = file.Update(context.Background(), tx, boil.Infer())
+	if err != nil {
+		if e := tx.Rollback(); e != nil {
+			return 0, fmt.Errorf("fatal: unable to rollback transaction on error: %s; %s", err.Error(), e.Error())
+		}
+	}
+
+	return uhassubmission.ID, nil
+}*/
+
+func DeleteUserSubmission(db *sql.DB, user_submission_id int) (int, error) {
```

Comment:
Hochgeladenes File sollte auch als gelöscht gekennzeichnet werden

Reply:
Wird in DeleteUserSubmissionHasFiles gekennzeichnet